examples/fileio/board/qemu_virt_aarch64/fileio.system

```diff
@@ -0,0 +1,295 @@
+
+        <map mr="blk_virtio_headers" vaddr="0x50_000_000" perms="rw" cached="false" setvar_vaddr="virtio_headers_vaddr" />
+        <map mr="blk_driver_metadata" vaddr="0x60_000_000" perms="rw" cached="false" setvar_vaddr="requests_vaddr" />
+
+        <irq irq="79" id="0" trigger="edge" />
+    </protection_domain>
+
+    <protection_domain name="BLK_VIRT" priority="101">
+        <program_image path="blk_virt.elf" />
+
+        <map mr="blk_driver_config" vaddr="0x40000000" perms="rw" cached="false" setvar_vaddr="blk_config_driver"     />
+        <map mr="blk_driver_request" vaddr="0x40200000" perms="rw" cached="false" setvar_vaddr="blk_req_queue_driver"  />
+        <map mr="blk_driver_response" vaddr="0x40400000" perms="rw" cached="false" setvar_vaddr="blk_resp_queue_driver" />
+        <map mr="blk_driver_data" vaddr="0x40600000" perms="rw" cached="false" setvar_vaddr="blk_data_driver" />
+        <setvar symbol="blk_data_driver_paddr" region_paddr="blk_driver_data" />
+
+        <map mr="blk_client_config" vaddr="0x30000000" perms="rw" cached="false" setvar_vaddr="blk_config"     />
+        <map mr="blk_client_request" vaddr="0x30200000" perms="rw" cached="false" setvar_vaddr="blk_req_queue"  />
+        <map mr="blk_client_response" vaddr="0x30400000" perms="rw" cached="false" setvar_vaddr="blk_resp_queue" />
+        <map mr="blk_client_data" vaddr="0x30600000" perms="rw" cached="false" setvar_vaddr="blk_client_data_start" />
+    </protection_domain>
+
+    <!-- FAT File system -->
+    <protection_domain name="fatfs" priority="100" pp="true">
+        <program_image path="fatfs.elf" />
+        <map mr="blk_client_config" vaddr="0x40_000_000" perms="r" cached="true" setvar_vaddr="config" />
+
+        <map mr="blk_client_request" vaddr="0x40_200_000" perms="rw" cached="true" setvar_vaddr="request" />
+        <map mr="blk_client_response" vaddr="0x40_400_000" perms="rw" cached="true" setvar_vaddr="response" />
+        <map mr="blk_client_data" vaddr="0x40_800_000" perms="rw" cached="true" setvar_vaddr="fs_metadata" />
+
+        <map mr="nfs_command_queue" vaddr="0x7_800_000" perms="rw" cached="true" setvar_vaddr="fatfs_command_queue" />
+        <map mr="nfs_completion_queue" vaddr="0x7_810_000" perms="rw" cached="true" setvar_vaddr="fatfs_completion_queue" />
+        <map mr="shared_fs_micropython" vaddr="0x7_900_000" perms="rw" cached="true" setvar_vaddr="client_data_addr"/>
+
+        <map mr="coroutine_stack1" vaddr="0xA0_000_000" perms="rw" cached="true" setvar_vaddr="coroutine_stack_one" />
+        <map mr="coroutine_stack2" vaddr="0xB0_000_000" perms="rw" cached="true" setvar_vaddr="coroutine_stack_two" />
+        <map mr="coroutine_stack3" vaddr="0xC0_000_000" perms="rw" cached="true" setvar_vaddr="coroutine_stack_three" />
+        <map mr="coroutine_stack4" vaddr="0xD0_000_000" perms="rw" cached="true" setvar_vaddr="coroutine_stack_four" />
```

Comment:
This should be global variables instead of memory regions in my opinion. The stacks should be small enough that having it as global data is fine.

Reply:
But I think by doing this way we would have guard pages to detect stack overflow from the coroutine stacks?